presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,472 @@
+                    typeStatement = "Nullable(" + typeStatement + ")";
+                }
+                columnList.add(format("%s %s", columnName, typeStatement));
+            }
+
+            ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+            ClickHouseTableProperties.getEngine(tableProperties)
+                    .ifPresent(value -> tableOptions.add("ENGINE =" + value));
+
+            String sql = format("CREATE TABLE %s.%s (%s) %s", remoteSchema, tableName, join(", ", columnList.build()),
+                    join(", ", tableOptions.build()));
+            execute(connection, sql);
+            return new JdbcOutputTableHandle(catalog, remoteSchema, remoteTable, columnNames.build(),
+                    columnTypes.build(), Optional.empty(), tableName);
+        }
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "CREATE DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn,
+            String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format("ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()),
+                    quoted(jdbcColumn.getColumnName()), quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            // clickhouse versions earlier than 8 do not support the above RENAME COLUMN
+            // syntax
+            if (ClickHouseErrorCode.UNKNOWN_EXCEPTION.toString().equals(e.getSQLState())) {
+                throw new PrestoException(NOT_SUPPORTED,
+                        format("Rename column not supported in catalog: '%s'", handle.getCatalogName()), e);
+            }
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected void copyTableSchema(Connection connection, String catalogName, String schemaName, String tableName,
+            String newTableName, List<String> columnNames)
+    {
+        String sql = format("CREATE TABLE %s AS  %s ", quoted(null, schemaName, newTableName),
+                quoted(null, schemaName, tableName));
+        execute(connection, sql);
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName,
+            SchemaTableName newTable)
+    {
+        if (!schemaName.equals(newTable.getSchemaName())) {
+            throw new PrestoException(NOT_SUPPORTED, "Table rename across schemas is not supported");
```

Comment:
Why? It's not supported in eg PostgreSQL because the syntax we use does not allow to provide new schema name.
Here you do provide new schema name.